tests/eas/preliminary.py

```diff
@@ -0,0 +1,241 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2016, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import json
+import time
+import re
+import pandas
+import StringIO
+
+from unittest import SkipTest
+
+from env import TestEnv
+from test import LisaTest
+
+"""
+Goal
+====
+
+Check that the configuration of a given device is suitable
+for running EAS.
+
+Detailed Description
+====================
+
+This test reads the kernel configuration and digs around in sysfs to
+check the following attributes are true:
+    * the minimum set of required config options are enabled
+    * all CPUs have access to the 'sched' CPUFreq governor
+    * energy aware scheduling is present and enabled
+
+Expected Behaviour
+==================
+
+All required config options are set, sched governor is present.
+
+"""
+
+TEST_CONF = {
+    'modules': ['cpufreq'],
+    'results_dir': 'PreliminaryTests',
+    'tools': [
+        'sysbench',
+    ]
+}
+
+class BasicCheckTest(LisaTest):
+    @classmethod
+    def setUpClass(cls):
+        cls.env = TestEnv(test_conf=TEST_CONF)
+        cls.target = cls.env.target
+
+class TestSchedGovernor(BasicCheckTest):
+    def test_sched_governor_available(self):
+        """
+        Check that the 'sched' or 'schedutil' cpufreq governor is available
+        """
+        fail_list = []
+        for cpu in self.target.list_online_cpus():
+            governors = self.target.cpufreq.list_governors(cpu)
+            if 'sched' not in governors and 'schedutil' not in governors:
+                fail_list.append(cpu)
+        msg = 'CPUs {} do not support sched[util] cpufreq governor'.format(
+            fail_list)
+        self.assertTrue(len(fail_list) == 0, msg=msg)
+
+class TestKernelConfig(BasicCheckTest):
+    def test_kernel_config(self):
+        """
+        Check that the kernel config has the basic requirements for EAS
+        """
+        kernel_config = self.target.config
+        if not kernel_config.text:
+            raise SkipTest('Kernel config not available on target')
+
+        # NB: We don't test for schedtune/schedutil, that's tested by
+        # TestSchedGovernor.
+        necessary_configs = [
+            # 'CONFIG_CPU_FREQ_STAT',
+            'CONFIG_CGROUPS',
+            'CONFIG_SMP',
+            'CONFIG_SCHED_MC',
+            'CONFIG_CPU_FREQ',
+            'CONFIG_CPU_IDLE',
+            'CONFIG_SCHED_DEBUG',
```

Comment:
Note that I've removed Schedtune stuf here too, let's test that separately since we'd probably like to be able to test kernels for EAS-related stuff without schedtune.